api/src/pcapi/core/users/api.py

```diff
@@ -469,6 +470,13 @@ def change_user_email(current_email: str, new_email: str) -> None:
     sessions = UserSession.query.filter_by(userId=current_user.id)
     repository.delete(*sessions)
 
+    UserEmailHistory.build_validation(
+        user=current_user,
+        old_email=current_email,
+        new_email=new_email,
+        device_id=device_id,
+    )
```

Comment:
Je ne vois pas où on ajoute cet objet au niveau de la base de données. Je suspecte que quelque chose l'ajoute à la session SQLAlchemy et commit, mais ça me paraît un peu magique. :) Je serais volontiers plus explicite ici, du genre : 

```suggestion
    email_history = UserEmailHistory.build_validation(
        user=current_user,
        old_email=current_email,
        new_email=new_email,
        device_id=device_id,
    )
    repository.save(email_history)
```

idem dans `request_email_update()` plus bas. Qu'en penses-tu ?

Reply:
je préfère quand le code est explicite

j'espère qu'Alambic est suffisament "intelligent" pour ne pas laisser la requete implicite lorsqu'on sauvegarde manuellement

Reply:
Je pense surtout que `UserEmailHistory` n'était pas enregistré (je ne vois pas de test vérifiant sa bonne création.
Je conseillerai d'ailleurs de déplacer ce bloc avec la modification du mail et de ne faire qu'un seul `repository.save`

Reply:
J'ai dis une bêtise, il y a bien un test, mais en passant via `user` du coup, ce serait interessant de vérifier que le `UserEmailHistory` a bien un `id` :)